Add json key validator

Abban opened the pull request:
We are about to add English localisation so we need to make sure that the different translation files stay in sync.

Ticket: https://phabricator.wikimedia.org/T256614

I decided to use PhpUnit for the following reasons:
* We are already comfortable with it
* We already have composer packages in this project
* It gIves us easy CLI output out of the box
* Easy to add more validations (eg can add new languages to the data providers)
* Less custom code to maintain

Abban committed:
Add json key validator

We are about to add English localisation so we
need to make sure that the different translation
files stay in sync.

Ticket: https://phabricator.wikimedia.org/T256614

Abban commented:
Note that CI is currently failing because of diffs between `messages` and `messages_laika`

gbirke commented on tests/Fixtures/JsonValidator.php:
```diff
@@ -0,0 +1,33 @@
+	}
+
+	public function assertJsonFilesHaveMatchingKeys( string $jsonFilePathA, $jsonFilePathB ): void {
+		$this->testCase->assertEqualsCanonicalizing(
```
The output of this comparison is really hard to read, because it shows the full content of the two arrays. Could you instead do two calls to `array_diff` with both sets of keys, expected the resulting array to equal an empty array (and a custom error message for the assertion to show what the result means)? This will show only keys that were added to one of the files but not the others.

gbirke commented:
I like the PHPUnit approach!

Abban committed:
Remove failing item from data provider

gbirke committed:
Use two-way array_diff for JSON key check

Check for missing and superfluous keys. This check assumes, that we'll
have one "default" set of messages that all other messages are compared
against.

gbirke committed:
Fix trigger configuration for Drone CI

The misspelled "triggers" (instead of "trigger") tried to run
development branches of the content.

gbirke reviewed (approved):
I've created a followup in #188 that does a more thorough check and fixes the failing Drone CI.
